frontend/src/app/projects/models/diagrams/model-diagram-dialog/model-diagram-dialog.component.ts

```diff
@@ -58,7 +70,7 @@ export class ModelDiagramDialogComponent {
     private dialogRef: MatDialogRef<ModelDiagramDialogComponent>,
     private dialog: MatDialog,
     @Inject(MAT_DIALOG_DATA)
-    public data: { modelSlug: string; projectSlug: string }
+    public data: { modelSlug: string; projectSlug: string; path: string }
```

Comment:
Here I would remove the path from the data, as we can easily use the method you currently use to set the path in here. To do so, I would add `path: string` at the top and in the constructor, we can then set it as follows:
`this.path = this.modelService.backendURLFactory(data.projectSlug, data.modelSlug)` (where we also need to add `private modelService: ModelService` to the constructor. In the html, we can then simply use `this.path` instead of `data.path`.